src/uibWinBuild.c

```diff
@@ -92,16 +96,19 @@ void app()
 			{
 				sensorParse(rxBuf[i]);
 			}
-
+			Serial_Write(globals.pOBC, (uint8_t *) &sensor_packet, sizeof(sensor_packet));
 		}
 
 		nchars = Serial_Read(globals.pOBC, rxBuf, 63);
 		if(nchars > 0)
 		{
-			if(nchars == 22)
+			memcpy(&tmp_buf[tmp_count], &rxBuf, nchars);
+			tmp_count += nchars;
+			if(tmp_count >= 22)
```

Comment:
What if we start from a partial message?  Please rewrite the `decodeStatusPacket` to work byte by byte.